src/OpeningHours.php

```diff
@@ -144,29 +144,32 @@ public static function createFromStructuredData(
 
     /**
      * @param  array  $data  hours definition array or sub-array
+     * @param  bool  $ignoreData  should ignore data
      * @param  array  $excludedKeys  keys to ignore from parsing
      * @return array
      */
-    public static function mergeOverlappingRanges(array $data, array $excludedKeys = ['data', 'dateTimeClass', 'filters', 'overflow']): array
+    public static function mergeOverlappingRanges(array $data, bool $ignoreData = true, array $excludedKeys = ['data', 'dateTimeClass', 'filters', 'overflow']): array
     {
         $result = [];
         $ranges = [];
 
-        foreach (static::filterHours($data, $excludedKeys) as $key => $value) {
+        foreach (static::filterHours($data, $excludedKeys) as $key => [$value, $data]) {
+            $dataString = json_encode($ignoreData ? null : $data);
```

Comment:
We can save a call to `json_encode` here:
```suggestion
            $dataString = $ignoreData ? '' : json_encode($data);
```

Reply:
Done